src/Core/ExecutionPathTracer/ExecutionPath.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace Microsoft.Quantum.IQSharp
+{
+    /// <summary>
+    /// Represents the qubit resources and operations traced out in an execution path of a Q# operation.
+    /// </summary>
+    public class ExecutionPath
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExecutionPathTracer"/> class.
+        /// </summary>
+        /// <param name="qubits">
+        /// An array of <c>QubitDeclaration</c>s that represents the declared qubits used in the execution path.
+        /// </param>
+        /// <param name="operations">
+        /// An array of <c>Operation</c>s that represents the operations used in the execution path.
+        /// </param>
+        public ExecutionPath(QubitDeclaration[] qubits, Operation[] operations)
+        {
+            this.Qubits = qubits;
+            this.Operations = operations;
+        }
+
+        /// <summary>
+        /// An array of <c>QubitDeclaration</c>s that represents the declared qubits used in the execution path.
+        /// </summary>
+        public QubitDeclaration[] Qubits { get; set; }
+
+        /// <summary>
+        /// An array of <c>Operation</c>s that represents the operations used in the execution path.
+        /// </summary>
+        public Operation[] Operations { get; set; }
+
+        /// <summary>
+        /// Serializes <c>ExecutionPath</c> into its JSON representation.
+        /// </summary>
+        /// <param name="prettyPrint">
+        /// Pretty prints the JSON (i.e. with white space and indents) if true (default = false).
+        /// </param>
+        public string ToJson(bool prettyPrint = false)
+        {
+            var options = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                IgnoreNullValues = true,
+                WriteIndented = prettyPrint,
+            };
+            return JsonSerializer.Serialize(this, options);
+        }
+    }
+
+    /// <summary>
+    /// Represents a qubit resource used in an execution path.
+    /// </summary>
+    public class QubitDeclaration
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QubitDeclaration"/> class.
+        /// </summary>
+        /// <param name="id">
+        /// Id of qubit.
+        /// </param>
+        /// <param name="numChildren">
+        /// Number of associated classical registers.
+        /// </param>
+        public QubitDeclaration(int id, int? numChildren = null)
+        {
+            this.Id = id;
+            this.NumChildren = numChildren;
+        }
+
+        /// <summary>
+        /// Id of qubit.
+        /// </summary>
+        public int Id { get; set; }
+
+        /// <summary>
+        /// Number of associated classical registers.
```

Comment:
Same question... if you can just use `0` as a default instead of `null`, it may be simpler, unless `0` and `null` have different meanings here.

Reply:
The `null` value here is so that this key doesn't get rendered in the JSON string. What are your thoughts on making it `null` so that it doesn't get rendered vs making it `0` for simplicity?

Reply:
Is there any case where `0` would mean something different than `null`? My preference would be to just use `0` and make the property non-nullable, as `NumChildren == null` doesn't seem very intuitive. You can just ignore this if it's `0` when parsing the JSON, correct?

Reply:
Yea, I agree. I think I was looking into this for `System.Text.Json` which I'm using, but there isn't a nice way _yet_ to ignore a property based on a value. I _think_ this is possible with `Newtonsoft`?